src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -656,6 +676,42 @@ internal override bool IsCallerMemberName
             }
         }
 
+        internal override int CallerArgumentExpressionParameterIndex
+        {
+            get
+            {
+                const WellKnownAttributeFlags flag = WellKnownAttributeFlags.IsCallerArgumentExpression;
+
+                if (!_packedFlags.TryGetWellKnownAttribute(flag, out bool value))
+                {
+                    var discardedUseSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
+                    bool isCallerArgumentExpression = !HasCallerLineNumberAttribute
+                        && !HasCallerFilePathAttribute
+                        && !HasCallerMemberNameAttribute
+                        && HasCallerArgumentExpressionAttribute
```

Comment:
>HasCallerArgumentExpressionAttribute [](start = 27, length = 36)

It doesn't feel usefull to record whether we have an attribute as a separate flag. It looks like it will be checked at most once and then, if the attribute is there, we will immediately search for it again. I would try to record a single flag with the meaning that the attribute is there and usable (the parameter type is right, source parameter can be located, etc.).

Reply:
@AlekseyTs Does this suggest keeping `PackedFlags` at its previous state? Wondering why this is different than other caller info attributes (if any difference)?

Reply:
>  Does this suggest keeping PackedFlags at its previous state? 

The saggestion is to add one flag instead of two.

> Wondering why this is different than other caller info attributes (if any difference)?

I don't know why it was decided to do that, but it appears that the result is affected merely by presence of other attributes, even when they are unusable. However, the new attribute doesn't affect any other attributes in this fashion.

---
In reply to: [600242044](https://github.com/dotnet/roslyn/pull/51952#discussion_r600242044) [](ancestors = 600242044)

Reply:
This is the issue that is still pending

> It doesn't feel usefull to record whether we have an attribute as a separate flag. It looks like it will be checked at most once and then, if the attribute is there, we will immediately search for it again. I would try to record a single flag with the meaning that the attribute is there and usable (the parameter type is right, source parameter can be located, etc.).

---
In reply to: [600068668](https://github.com/dotnet/roslyn/pull/51952#discussion_r600068668) [](ancestors = 600068668)